src/sql/src/kafka_util.rs

```diff
@@ -34,16 +34,42 @@ enum ValType {
     // Number with range [lower, upper]
     Number(i32, i32),
     Boolean,
+    EnvVar,
+}
+
+impl ValType {
+    fn process_val(&self, val: &Value) -> Result<String, anyhow::Error> {
+        Ok(match (&self, val) {
+            (ValType::String, Value::String(v)) => v.to_string(),
+            (ValType::Boolean, Value::Boolean(b)) => b.to_string(),
+            (ValType::Path, Value::String(v)) => {
+                if std::fs::metadata(&v).is_err() {
+                    bail!("file does not exist")
+                }
+                v.to_string()
+            }
+            (ValType::Number(lower, upper), Value::Number(n)) => match n.parse::<i32>() {
+                Ok(parsed_n) if *lower <= parsed_n && parsed_n <= *upper => n.to_string(),
+                _ => bail!("must be a number between {} and {}", lower, upper),
+            },
+            (ValType::EnvVar, Value::String(v)) => {
+                std::env::var(v).map_err(|_| anyhow!("environment variable not set"))?
```

Comment:
nit: technically this is either "the env var doesn't exist" or "the env var is invalid utf-8." To be perfectly correct we should handle the cases differently.